compiler/hash-typecheck/src/new/passes/symbol_resolution.rs

```diff
@@ -48,6 +51,75 @@ impl<'tc> SymbolResolutionPass<'tc> {
     }
 }
 
+impl<'tc> SymbolResolutionPass<'tc> {
+    /// Run a function for each stack member in the given pattern.
+    ///
+    /// The stack members are found in the `AstInfo` store, specifically the
+    /// `stack_members` map. They are looked up using the IDs of the pattern
+    /// binds, as added by the `add_stack_members_in_pat_to_buf` method of the
+    /// `ScopeDiscoveryPass`.
+    fn for_each_stack_member_of_pat(
+        &self,
+        node: ast::AstNodeRef<ast::Pat>,
+        f: impl Fn(StackMemberId) + Copy,
+    ) {
+        let for_spread_pat = |spread: &ast::AstNode<ast::SpreadPat>| {
+            if let Some(name) = &spread.name {
+                if let Some(member_id) =
+                    self.ast_info().stack_members().get_data_by_node(name.ast_ref().id())
+                {
```

Comment:
could use a `let_chain` here:

```suggestion
            if let Some(name) = &spread.name && let Some(member_id) =
                    self.ast_info().stack_members().get_data_by_node(name.ast_ref().id())
                {
```